### PR TITLE
A throughput vs. CPU benchmark.

### PR DESCRIPTION
**Pull request by coryan**
Introduce a new benchmark to measure the throughput when uploading and
downloading objects, as well as the CPU cost (think "cpu cycles per
uploaded byte") to achieve that throughput. We will use this benchmark
to evaluate some performance improvements I have in mind.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2695)

### CODE REVIEW

File: google/cloud/storage/benchmarks/benchmark_utils.h [hunk 1]
```diff
@@ -79,6 +82,27 @@ std::string BuildUsage(std::vector<OptionDescriptor> const& desc,
...
                                       std::vector<std::string> argv);
 
+class SimpleTimer {
+ public:
+  SimpleTimer() = default;
+
+  void Start();
+  void Stop();
+
+  std::chrono::microseconds elapsed_time() const { return elapsed_time_; }
```
Comment (devbww) [inline google/cloud/storage/benchmarks/benchmark_utils.h]: Perhaps a note that these are only valid after `Stop()`?

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+  return 0;
+} catch (std::exception const& ex) {
+  std::cerr << "Standard exception raised: " << ex.what() << "\n";
+  return 1;
+}
+
+namespace {
+char const* ToString(OpType type) {
+  static char const* kOpTypeNames[] = {"UPLOAD", "DOWNLOAD", "LAST"};
```
Comment (devbww) [inline google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc]: Is just `switch (type) ...` easier?

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+
+  std::uniform_int_distribution<std::uint64_t> size_generator(
+      options.minimum_object_size, options.maximum_object_size);
+  std::uniform_int_distribution<std::uint64_t> chunk_generator(
+      options.minimum_chunk_size, options.maximum_chunk_size);
+
+  auto deadline = std::chrono::steady_clock::now() + options.duration;
+
+  gcs_bm::SimpleTimer timer;
+  // This obviously depends on the size of the objects, but a good estimate is
```
Comment (devbww) [inline google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc]: s/ is$//

File: google/cloud/storage/benchmarks/benchmark_utils.h (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/benchmarks/benchmark_utils.h]: Done.

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc]: Done.

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc (hunk 3 above)
Comment (coryan) [reply google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc]: Fixed.

**Pull request merged**